File: api/spring-boot-api/src/main/java/com/spicytodo/model/Todo.java
package com.spicytodo.model;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Todo {
    private String id;

    @NotBlank(message = "Todo text is required")
    @Size(min = 1, max = 500, message = "Todo text must be between 1 and 500 characters")
    private String text;

    private Priority priority = Priority.medium;
    private Boolean completed = false;
    private String dueDate;
    private String reminderTime;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    public Todo(String text, Priority priority, Boolean completed, String dueDate, String reminderTime) {
        this.id = UUID.randomUUID().toString();
        this.text = text;
        this.priority = priority != null ? priority : Priority.medium;
        this.completed = completed != null ? completed : false;
        this.dueDate = dueDate;
        this.reminderTime = reminderTime;
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    public void updateTimestamp() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: api/dotnet-api/Models/Todo.cs
using System.ComponentModel.DataAnnotations;

namespace DotnetApi.Models
{
    public class Todo
    {
        public string Id { get; set; } = Guid.NewGuid().ToString();

        [Required(ErrorMessage = "Todo text is required")]
        [StringLength(500, MinimumLength = 1, ErrorMessage = "Todo text must be between 1 and 500 characters")]
        public string Text { get; set; } = string.Empty;

        public Priority Priority { get; set; } = Priority.medium;
        public bool Completed { get; set; } = false;
        public string? DueDate { get; set; }
        public string? ReminderTime { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        public void UpdateTimestamp()
        {
            UpdatedAt = DateTime.UtcNow;
        }
    }
}

